integration-tests/src/test/java/org/apache/fineract/integrationtests/StretchyReportTest.java

```diff
@@ -0,0 +1,108 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import org.apache.fineract.integrationtest.stretchyreports.StretchyReportHelper;
+import org.apache.fineract.integrationtests.common.ClientHelper;
+import org.apache.fineract.integrationtests.common.CommonConstants;
+import org.apache.fineract.integrationtests.common.GlobalConfigurationHelper;
+import org.apache.fineract.integrationtests.common.Utils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StretchyReportTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StretchyReportTest.class);
+    private RequestSpecification requestSpec;
+    private ResponseSpecification responseSpec;
+    private StretchyReportHelper stretchyReportHelper;
+    private static final String STRETCHY_GET_REPORT_URL = "/fineract-provider/api/v1/reports";
+    private static final String STRETCHY_REPORT_URL = "/fineract-provider/api/v1/runreports";
+
+    @BeforeEach
+    public void setup() {
+        Utils.initializeRESTAssured();
+        this.requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        this.requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        this.responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+    }
+
+    @Test
+    public void testReportPagination() {
+        this.stretchyReportHelper = new StretchyReportHelper(this.requestSpec, this.responseSpec);
+
+        final ResponseSpecification errorResponse = new ResponseSpecBuilder().expectStatusCode(400).build();
+        StretchyReportHelper validationErrorHelper = new StretchyReportHelper(this.requestSpec, errorResponse);
+        for (int i = 0; i < 20; i++) {
+            final Integer clientID = ClientHelper.createClient(this.requestSpec, this.responseSpec);
+            ClientHelper.verifyClientCreatedOnServer(this.requestSpec, this.responseSpec, clientID);
+        }
+        String url = STRETCHY_REPORT_URL + "/" + "Client Listing" + "?" + Utils.TENANT_IDENTIFIER + "&R_officeId=1";
+        LinkedHashMap reportData = this.stretchyReportHelper.getStretchyReportDetail(this.requestSpec, this.responseSpec, url, "");
+        ArrayList<String> rdata = (ArrayList<String>) reportData.get("data");
+        Integer reportDataSize = rdata.size();
+        Assertions.assertNotNull(reportDataSize);
+
+        Boolean isPaginationAllowed = true;
+        if (isPaginationAllowed) {
+            Integer pageNo = 0;
+            Integer orderBy = 1;
+            Integer pageCount = 0;
+            Integer pageContent = Math.toIntExact(
+                    GlobalConfigurationHelper.getGlobalConfigurationByName(requestSpec, responseSpec, "page-content-limit").getValue());
+            pageCount = reportDataSize / pageContent;
```

Comment:
This logic is very much not trivial. Do you mind explaining whats going on here?

Reply:
this code help to get total pageNo count as per the pageContent value.